src/modules/aws-ec2/iam-instance-profile/sub-modules/diassociateIamInstanceProfile.ts

```diff
@@ -0,0 +1,20 @@
+import { EC2 } from "aws-sdk";
+import { describeIamInstanceProfileAssociations } from "./describeIamInstanceProfileAssociations";
+import { logger } from "../../../../logger";
+
+export const diassociateIamInstanceProfile = async (instanceId: string) => {
+  const ec2 = new EC2();
+  const responseData = await describeIamInstanceProfileAssociations(instanceId);
+  if (!responseData.IamInstanceProfileAssociations) {
+    return;
+  }
+  const profileAssociations = responseData.IamInstanceProfileAssociations;
+  for (const profileAssociation of profileAssociations) {
+    const associationId = profileAssociation.AssociationId || "";
+    const params = {
+      AssociationId: associationId
+    };
+    await ec2.disassociateIamInstanceProfile(params).promise();
```

Comment:
How this function behaves if `AssociationId` is `""`?

Reply:
Yeah looks like it will throw an error. Changed this to:
```
  for (const profileAssociation of profileAssociations) {
    if (!profileAssociation.AssociationId) {
      break;
    }
    const params: DisassociateIamInstanceProfileRequest = {
      AssociationId: profileAssociation.AssociationId
    };
    await ec2.disassociateIamInstanceProfile(params).promise();
  }
```